Generators/include/Generators/GeneratorHepMCParam.h

```diff
@@ -30,8 +30,7 @@ namespace eventgen
  **/
 
 struct GeneratorHepMCParam : public o2::conf::ConfigurableParamHelper<GeneratorHepMCParam> {
-  std::string fileName = "";
-  int version = 2;
```

Comment:
We have a problem here: `HepMC.filename` is used in existing examples and tests. The CI fails since you take this away. Also, there seems to be an inconsistency in the description above: In one point you say that `.filename` is no longer supported ... but then `filename ` occurs in newly added examples. It would be best to avoid changing things too much...  so better keep the existing keys and functionality. I would also like to encourage a review from @preghenella as original author on this.

Reply:
Hi Sandro, 

Yes, the newly added documentation is a bit out of sync - but note the **In the near future** section.   This was because I did not know which MR you would take.    I've added code to `GeneratorHepMC` which will throw a `fatal` if the old key is used, and gives an appropriate message about what to do instead. 

Note that the previous code only allowed one to read from 1 file, and that you had to know the version (and specify it) of the format.   The new code allows for multiple input files and the it figures out the format (including version) itself.   It also supports all the file formats supported by the HepMC3 install (ROOT, LHEF, HepEVT, remote access ROOT, etc. are some of the options in addition to plain ASCII HepMC). 

I can update the docs and add them to the MR.

Reply:
Fixed docs and other places where `HepMC.fileName` was used.